### PR TITLE
chore: unify template naming

### LINKED ISSUE #15754: Unify template naming
We currently have very inconsistent naming in our [templates](https://github.com/coder/coder/tree/main/examples/templates) directory (e.g `aws-devcontainer` and `devcontainer-kubernetes`).

I will open PRs to fix this inconsistency.

### PR DESCRIPTION
**Pull request by phorcys420**
This PR changes template names and docs to follow the `<provider>-<os/whatever>` format for all templates.
I've decided not to split this into multiple PRs because I'd have to edit rebase the other PRs once one of them gets merged, this should be relatively low-impact anyways.

This aligns with our goals to make templates more user-friendly.

Closes #15754

### CODE REVIEW

File: docs/admin/templates/managing-templates/devcontainers/add-devcontainer.md [hunk 1]
```diff
@@ -122,8 +122,8 @@ their development environments:
...
 | ------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| [Docker dev containers](https://github.com/coder/coder/tree/main/examples/templates/devcontainer-docker)            | Docker provisions a development container.                                                                                                                          |
-| [Kubernetes dev containers](https://github.com/coder/coder/tree/main/examples/templates/devcontainer-kubernetes)    | Provisions a development container on the Kubernetes cluster.                                                                                                       |
+| [Docker dev containers](https://github.com/coder/coder/tree/main/examples/templates/docker-devcontainer)            | Docker provisions a development container.                                                                                                                          |
```
[the remark this answers is not in the document]
Comment (phorcys420) [reply docs/admin/templates/managing-templates/devcontainers/add-devcontainer.md:125]: This is normal, the link will be available after the PR gets merged.

File: docs/admin/templates/managing-templates/devcontainers/add-devcontainer.md [hunk 2]
```diff
@@ -122,8 +122,8 @@
...
+| [Kubernetes dev containers](https://github.com/coder/coder/tree/main/examples/templates/kubernetes-devcontainer)    | Provisions a development container on the Kubernetes cluster.                                                                                                       |
```
[the remark this answers is not in the document]
Comment (phorcys420) [reply docs/admin/templates/managing-templates/devcontainers/add-devcontainer.md:126]: This is normal, the link will be available after the PR gets merged.

File: docs/admin/templates/extending-templates/docker-in-workspaces.md [hunk 3]
```diff
@@ -148,7 +148,7 @@ nodes. Refer to sysbox's
 to ensure your nodes are compliant.
 
 To get started with `envbox` check out the
-[starter template](https://github.com/coder/coder/tree/main/examples/templates/envbox)
+[starter template](https://github.com/coder/coder/tree/main/examples/templates/kubernetes-envbox)
```
[the remark this answers is not in the document]
Comment (phorcys420) [reply docs/admin/templates/extending-templates/docker-in-workspaces.md:151]: This is normal, the link will be available after the PR gets merged.

**Comment by phorcys420** [comment]
I don't know why all these checks are failing, I'll take a look tomorrow.

**Comment by matifali** [comment]
You need to run `make gen` too. 
Also `make fmt` for any possible formatting issues.

**Review by matifali — APPROVED** [review]
LGTM. 
It could be a breaking change if someone is initializing templates in CI. 

@bpmct fir visibility.

**Comment by matifali** [comment]
@phorcys420 you may also need to run `make update-golden-files` as we use the template ids for some CLI tests.
See: https://github.com/coder/coder/actions/runs/12176090929/job/33961115484?pr=15757#step:7:579

**Review by hugodutka — COMMENTED** [review]
We've started tracking usage of template examples in telemetry with the current coder release. The PR was merged last week: https://github.com/coder/coder/pull/15671. We use the example IDs to identify the templates in telemetry, so changing them will confuse the statistics. It's fine if we change the IDs now, but they should remain static in the future. @phorcys420, could you add the following comment in `examples/examples.go` above the `//go:embed` directives?

> Template paths should not change over time. They are used as persistent identifiers in telemetry.

**Comment by phorcys420** [comment]
@hugodutka thanks for pointing this out!
I guess we have good timing then! I wanted to do this to set a standard naming convention to avoid confusing our users so I think it will also help you if the names are standardized.

**Comment by phorcys420** [comment]
@matifali is there a way we can merge without weekly-docs passing?

**Comment by phorcys420** [comment]
whoops

**Pull request merged**